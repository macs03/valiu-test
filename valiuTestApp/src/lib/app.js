import SocketLoader from './socket-loader';

const globalLoaders = [SocketLoader];

export default class App {
  static socket = undefined;

  static boot() {
    return Promise.all(globalLoaders.map((s) => s.boot(App)))
      .then(() => {
        console.info('App is ready!!!');
      })
      .catch((err) => console.error(`Error while booting the app`, err));
  }
}
