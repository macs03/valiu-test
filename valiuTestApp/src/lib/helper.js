const ramdom = (max) => {
  return Math.floor(Math.random() * Math.floor(max));
};

export default ramdom;
