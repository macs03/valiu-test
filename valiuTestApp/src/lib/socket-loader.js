import io from 'socket.io-client';
import config from './config';

export default class SocketLoader {
  static boot(App) {
    const server = `${config.server}:${config.port}`;
    const socket = io(server, {
      transports: ['websocket'],
      jsonp: false,
    });

    socket.on('connect', () => console.info('Connection Sucessfull!!'));
    App.socket = socket;
  }
}
