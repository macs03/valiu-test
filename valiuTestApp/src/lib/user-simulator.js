/**
 * Each "interaction" is a series of methods (props on a component) React Native would call
 * when a real user using a real device uses the application. For instance: doing a PRESS
 * on a component is actually calling the onPressIn -> onPressOut -> onPress props
 * in sequence with the proper arguments
 */
export const Interactions = {
  PRESS: ['onPressIn', 'onPressOut', 'onPress'],
  TEXT_INPUT: ['onChangeText', 'onChange'],
  FOCUS_INPUT: ['onFocus'],
  UNFOCUS_INPUT: ['onBlur'],
  SWIPE_RIGHT: [
    'onStartShouldSetResponder',
    'onResponderGrant',
    'onResponderRelease',
    'onHandlerStateChange',
  ],
  SWIPE_LEFT: [
    'onStartShouldSetResponder',
    'onResponderGrant',
    'onResponderRelease',
    'onHandlerStateChange',
  ],
  SWITCH: ['onAsyncPress'],
  SNAP_TO_INDEX: ['onSnapToItem'],
  TAP: ['onHandlerStateChange'],
};

/**
 * This methods returns the arguments for a given "interaction prop", so that the
 * component receives the parameters it's expecting as if an actual user using a real
 * device is interacting with it
 */
const eventFor = (gesture, method, arg) => {
  const mapping = {
    onChangeText: arg,
    onChange: {
      nativeEvent: {
        text: arg,
      },
    },
    onSnapToItem: arg,
    onHandlerStateChange: arg,
    onAsyncPress: arg,
  };

  mapping[Interactions.SWIPE_RIGHT] = {
    onResponderGrant: {
      nativeEvent: {
        pageX: arg,
      },
    },
    onResponderRelease: {
      nativeEvent: {
        pageX: 2 * arg,
      },
    },
    onHandlerStateChange: {
      nativeEvent: {
        state: 4,
        pageX: arg,
      },
    },
  };
  mapping[Interactions.SWIPE_LEFT] = {
    onResponderGrant: {
      nativeEvent: {
        pageX: arg,
      },
    },
    onResponderRelease: {
      nativeEvent: {
        pageX: 2 * arg,
      },
    },
    onHandlerStateChange: {
      nativeEvent: {
        state: 4,
        pageX: arg,
      },
    },
  };
  mapping[Interactions.TAP] = {
    onResponderGrant: {
      nativeEvent: {
        pageX: arg,
      },
    },
    onResponderRelease: {
      nativeEvent: {
        pageX: 2 * arg,
      },
    },
    onHandlerStateChange: {
      nativeEvent: {
        state: 4,
        pageX: arg,
      },
    },
  };

  return (mapping[gesture] || mapping)[method] || mapping[method];
};

/**
 * This methods enables the `on(component).simulate(PRESS)` syntax. it basically creates a
 * simulate method that calls a series of prop functions in sequence, the same way React Native would
 * when a user interaction is recognized
 */
export const on = (component) => {
  if (!component) {
    throw new Error(`Can't simulate gestures on null or undefined`);
  }
  return {
    simulate: (gesture, arg) => {
      gesture
        .filter((method) => component.props[method])
        .forEach((method) => {
          component.props[method](eventFor(gesture, method, arg));
        });
    },
  };
};

/**
 * This methods returns the element for the corresponding testId
 */
export const findByTestId = (tree, testID) => {
  if (tree) {
    return tree.root.findByProps({testID});
  }
};
