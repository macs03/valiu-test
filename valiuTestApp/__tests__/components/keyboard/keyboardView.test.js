/**
 * @format
 */

import 'react-native';
import React from 'react';
import KeyboardView from '../../../src/components/keyboard/keyboardView';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import {on, Interactions, findByTestId} from '../../../src/lib/user-simulator';

it('renders correctly', () => {
  const tree = renderer.create(
    <KeyboardView handlingPad={() => {}} title={'Test'} />,
  );
  expect(tree).toMatchSnapshot();
});

it('calls on handlingPad button', () => {
  const spy = jest.fn();
  const tree = renderer.create(
    <KeyboardView handlingPad={spy} title={'Test'} />,
  );

  const button = findByTestId(tree, '3');

  on(button).simulate(Interactions.PRESS);

  expect(spy).toHaveBeenCalled();
});
