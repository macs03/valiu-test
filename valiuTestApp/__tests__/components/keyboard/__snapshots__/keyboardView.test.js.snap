// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgba(25,105,0,0.8)",
      "bottom": 0,
      "height": "45%",
      "justifyContent": "center",
      "left": 0,
      "paddingBottom": 70,
      "position": "absolute",
      "width": "100%",
    }
  }
>
  <Text
    style={
      Object {
        "alignSelf": "center",
        "color": "white",
      }
    }
  >
    Test
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "flexWrap": "wrap",
        "justifyContent": "space-around",
        "paddingTop": 20,
        "width": "100%",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#696969",
          "borderRadius": 25,
          "height": "33%",
          "justifyContent": "center",
          "opacity": 1,
          "width": "33%",
        }
      }
      testID="0"
    >
      <Text>
        0
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#696969",
          "borderRadius": 25,
          "height": "33%",
          "justifyContent": "center",
          "opacity": 1,
          "width": "33%",
        }
      }
      testID="1"
    >
      <Text>
        1
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#696969",
          "borderRadius": 25,
          "height": "33%",
          "justifyContent": "center",
          "opacity": 1,
          "width": "33%",
        }
      }
      testID="2"
    >
      <Text>
        2
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#696969",
          "borderRadius": 25,
          "height": "33%",
          "justifyContent": "center",
          "opacity": 1,
          "width": "33%",
        }
      }
      testID="3"
    >
      <Text>
        3
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#696969",
          "borderRadius": 25,
          "height": "33%",
          "justifyContent": "center",
          "opacity": 1,
          "width": "33%",
        }
      }
      testID="4"
    >
      <Text>
        4
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#696969",
          "borderRadius": 25,
          "height": "33%",
          "justifyContent": "center",
          "opacity": 1,
          "width": "33%",
        }
      }
      testID="5"
    >
      <Text>
        5
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#696969",
          "borderRadius": 25,
          "height": "33%",
          "justifyContent": "center",
          "opacity": 1,
          "width": "33%",
        }
      }
      testID="6"
    >
      <Text>
        6
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#696969",
          "borderRadius": 25,
          "height": "33%",
          "justifyContent": "center",
          "opacity": 1,
          "width": "33%",
        }
      }
      testID="7"
    >
      <Text>
        7
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#696969",
          "borderRadius": 25,
          "height": "33%",
          "justifyContent": "center",
          "opacity": 1,
          "width": "33%",
        }
      }
      testID="8"
    >
      <Text>
        8
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#696969",
          "borderRadius": 25,
          "height": "33%",
          "justifyContent": "center",
          "opacity": 1,
          "width": "33%",
        }
      }
      testID="9"
    >
      <Text>
        9
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#696969",
          "borderRadius": 25,
          "height": "33%",
          "justifyContent": "center",
          "opacity": 1,
          "width": "33%",
        }
      }
      testID=","
    >
      <Text>
        ,
      </Text>
    </View>
  </View>
</View>
`;
